### DISCUSSION TITLE
Configuration: different level for src and test

### DISCUSSION THREAD

**Discussion by tamuhey** [microsoft/pyright#2068] [Q&A] [answered]
Hi,
I have a project whose structure is roughly as follows:

```
src/  # source files
test/ # test files
pyrightconfig.json
pyproject.toml
```

I want to use `strict` mode for `src`, and `basic` mode for `test`. What should I do?

**Comment by erictraut** [comment, marked as answer]
Include the following in your `pyrightconfig.json` file:

```
"strict": [
    "src"
]
```

**Reply by tamuhey** [reply]
@erictraut  Thanks! If I want to enable a single diagnostic rule only for `src`, what should I do? 
e.g. I want to do like below:

```
"src": {
 "reportMissingTypeStub": "error"
}
```

**Reply by erictraut** [reply]
If you want to override a diagnostic rule for a single file, you cannot do so in the config file. You would need to use a `# pyright: reportMissingTypeStub=error` or `# pyright: reportMissingTypeStub=true` comment at the top of that individual source file to override the diagnostic setting in the config file.

**Reply by erictraut** [reply]
For more details about how this works, refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/configuration.md).

**Reply by Avasam** [reply]
The workaround I've used so far is to have different `pyrightconfig.*.json` with their own include (and exclude duplicating the include of other configs), and on the CI run the pyright test step once per config.
It's also the workaround used by typeshed.
This, however, causes a lot of configuration duplication, and prevents Pylance from used to correct configuration and showing the right errors for all those folders configured differently. (see: https://github.com/microsoft/pylance-release/discussions/3426) . It also causes failures from later CI jobs to be hidden if an earlier job failed.
